test/boost/sstable_compaction_test.cc

```diff
@@ -5017,3 +5017,44 @@ SEASTAR_TEST_CASE(compaction_manager_stop_and_drain_race_test) {
     testlog.info("stopping compaction manager");
     co_await cm.stop();
 }
+
+SEASTAR_TEST_CASE(move_sstables_from_staging_test) {
+    return test_env::do_with_async([](test_env &env) {
+        auto builder = schema_builder("tests", "table_move_sstables_from_staging_test")
+                .with_column("id", utf8_type, column_kind::partition_key)
+                .with_column("value", int32_type);
+        // Such that no compaction is triggered when adding sstables to the table.
+        builder.set_compaction_strategy(sstables::compaction_strategy_type::null);
+        auto s = builder.build();
+
+        auto tmp = tmpdir();
+        auto staging_path = fs::path(tmp.path().string()) / fs::path(sstables::staging_dir);
+        testlog.info("staging_path={}", staging_path.c_str());
+        touch_directory(staging_path.c_str()).get();
+
+        table_for_tests cf(env.manager(), s, tmp.path().string());
```

Comment:
s/string/native/

Reply:
and more below

Reply:
doc for std::filesystem::path::string says: 
"Returns the internal pathname in native pathname format, converted to specific string type."

whereas path::native:
2) Returns the native-format representation of the pathname by reference.

we've been using tmp.path().string(), but native() also works and can avoid a copy. will switch to native().